lrcget: use cargo-tauri.hook

TomaSajt opened the pull request:
Things done:
- combine the frontend build into the main build
- use `cargo-tauri.hook` to generate `.desktop` file and icons
  - note: this changed the name of the executable to be all uppercase
    - I edited `meta.mainProgram` to reflect this
- reorder dependencies by category and sort each category alphabetically
- use `dontWrapGApps` to avoid double wrapping the binary

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TomaSajt committed:
lrcget: use cargo-tauri.hook

TomaSajt commented on pkgs/by-name/lr/lrcget/package.nix:
```diff
@@ -1,69 +1,73 @@
     owner = "tranxuanthang";
     repo = "lrcget";
-    rev = "${version}";
+    tag = version;
```
One benefit is that it only works with tags and not branches.
(It's equivalent to doing `rev = "refs/tags/${tag}"`

0x61nas replied:
ah, i see

TomaSajt commented on pkgs/by-name/lr/lrcget/package.nix:
```diff
@@ -112,7 +111,7 @@ rustPlatform.buildRustPackage rec {
       Scrumplex
     ];
-    mainProgram = "lrcget";
+    mainProgram = "LRCGET";
```
It's a breaking change, but only if you were using it from the command line directly.

If you use the `.desktop` file, it shouldn't matter what the binary is called, it's essentially just an implementation detail.

If you use `nix run .#lrcget` it should respect the new mainProgram.

0x61nas replied:
I would prefer to keep it as it is, though, because caps names are silly and hard to write/read(humble personal opinion). Plus, if its unnecessary, then why bother?

TomaSajt replied:
Okay so: if you don't use cargo-tauri.hook, you might get a different binary name then if you used cargo-tauri.hook.
So assuming we want to use cargo-tauri.hook, we need to set mainProgram to the new binary name.

I could add `ln -s $out/bin/LRCGET $out/bin/lrcget`
to make an alias, though, if you want me to.

0x61nas replied:
So, this is kinda necessary at least by my understanding

0x61nas replied:
> I could add ln -s $out/bin/LRCGET $out/bin/lrcget
to make an alias, though, if you want me to.

no, we can go with the caps as it's a requirement, and leave the aliasesing for the user then

Scrumplex commented on pkgs/by-name/lr/lrcget/package.nix:
```diff
@@ -76,30 +80,25 @@ rustPlatform.buildRustPackage rec {
-    })
-  ];
+  # don't automatically wrap binary to avoid double wrapping it
+  dontWrapGApps = true;
+
+  postFixup =
+    # WEBKIT_DISABLE_COMPOSITING_MODE essential in NVIDIA + compositor https://github.com/NixOS/nixpkgs/issues/212064#issuecomment-1400202079
+    lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+      wrapGApp "$out/bin/LRCGET" \
+        --set WEBKIT_DISABLE_COMPOSITING_MODE 1
```
Why not

```nix
    gappsWrapperArgs+=(
      # WEBKIT_DISABLE_COMPOSITING_MODE essential in NVIDIA + compositor https://github.com/NixOS/nixpkgs/issues/212064#issuecomment-1400202079
      --set WEBKIT_DISABLE_COMPOSITING_MODE 1
    )
```

instead?

TomaSajt replied:
True, I'll change it once I get back home

Scrumplex commented:
Other than my comment above, this looks good! :D

TomaSajt commented:
Pushed a commit that addresses @Scrumplex's review.
It also contains changes to buildInputs: I removed `dbus` and `gtk3` (not sure why `dbus` was needed, and `gtk3` is propagated anyways) and moved `openssl` to the linux-only deps (it's not needed on darwin).
